src/assets/javascript/main.js

```diff
@@ -0,0 +1,65 @@
+import "../../assets/main.css"; 
+const button = require("../../components/button");
+const inputReuslt = document.querySelector(".js-result");
+
+//-------------------- textContent of all buttons-----------------------------//
+
+const buttonComponents = [
```

Comment:
Ok, how can we improve this button list,  to avoid duplicate a lot of button components?

Reply:
i tried using `Array.filter()` and looks like it's working
i'll commit later, just one minute

Reply:
I don't mean array filter

Reply:
think a bit more :)

Reply:
You can say that solution, before you start to implement ;)

Reply:
> You can say that solution, before you start to implement ;)

ok

Reply:
> Ok, how can we improve this button list, to avoid duplicate a lot of button components?

but how "it's not just don't duplicate" 
ah and other thing, that function that i just created,it's not working for the all buttons :(

Reply:
> Ok, how can we improve this button list, to avoid duplicate a lot of button components?

do you mean i call the same function for different buttons?
like: `
const buttonComponents = [
	button("C","js-clearAll","js-upper"),button("+/-","js-lessMore","js-upper"),button("«","js-del","js-upper"),button("%","js-buttons","js-upper"),
	button("7","js-buttons",""),button("8","js-buttons",""),button("9","js-buttons",""),button("/","js-buttons",""),
	button("4","js-buttons",""),button("5","js-buttons",""),button("6","js-buttons",""),button("*","js-buttons",""),
	button("1","js-buttons",""),button("2","js-buttons",""),button("2","js-buttons",""),button("3","js-buttons",""),button("+","js-buttons",""),
	button("0","js-buttons","js-bottom"),button(".","js-buttons","js-bottom"),button("=","js-equal","js-bottom"),button("-","js-buttons","js-bottom"),
];
`

Reply:
nope